mirrord/agent/src/runtime/crio.rs

```diff
@@ -0,0 +1,100 @@
+use bytes::Bytes;
+use futures::TryFutureExt;
+use http::{Request, Response};
+use http_body_util::{BodyExt, Empty};
+use hyper::{body::Incoming, client::conn};
+use k8s_cri::v1alpha2::{runtime_service_client::RuntimeServiceClient, ContainerStatusRequest};
+use tokio::net::UnixStream;
+use tonic::transport::{Endpoint, Uri};
+use tower::service_fn;
+use tracing::error;
+
+use crate::{
+    error::{AgentError, Result},
+    runtime::{ContainerInfo, ContainerRuntime},
+};
+
+static CRIO_DEFAULT_SOCK_PATH: &str = "/host/run/crio/crio.sock";
+
+#[derive(Debug)]
+pub(crate) struct CriOContainer {
+    pub container_id: String,
+}
+
+impl CriOContainer {
+    pub fn from_id(container_id: String) -> Self {
+        CriOContainer { container_id }
+    }
+
+    async fn request(path: String) -> Result<Response<Incoming>> {
+        let stream = UnixStream::connect(CRIO_DEFAULT_SOCK_PATH).await?;
+        let (mut request_sender, connection) = conn::http1::handshake(stream).await?;
```

Comment:
can't we use `reqwest` or something more high level here?

Reply:
We have hyper 1.0.0-rc and there is no rc version for reqwest and I don't want issues with error handling